### PR TITLE
CI fixes

### PR DESCRIPTION
**Pull request by LisannaAtGalois**

### CODE REVIEW

**Comment by LisannaAtGalois** [comment]
Tested by manually triggering the workflow via:

```bash
curl \
  -u LisannaAtGalois \
  -X POST \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/GaloisInc/saw-script/actions/workflows/build.yml/dispatches \
  -d '{"ref":"lisanna/manual-release-artifacts", "inputs":{"releaseArtifacts":"1"}}'
```

There's an issue with building cryptol on 8.6.5, will investigate more. https://github.com/GaloisInc/saw-script/runs/1908547733?check_suite_focus=true

File: .github/ci.sh [hunk 1]
```diff
@@ -120,15 +120,13 @@ build() {
   ghc_ver="$(ghc --numeric-version)"
   cp cabal.GHC-"$ghc_ver".config cabal.project.freeze
   cabal v2-update
-  echo "allow-newer: all" >> cabal.project.local
```
Comment (jared-w) [inline .github/ci.sh:123]: Yisss :heart:

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -25,7 +25,6 @@ jobs:
   docker-saw:
     runs-on: ubuntu-18.04
     needs: [outputs]
-    continue-on-error: true
```
Comment (jared-w) [inline .github/workflows/nightly.yml:28]: Another big win!

**Review by jared-w — APPROVED** [review]
Very excited for these. Nice work! :)

**Review by LisannaAtGalois — COMMENTED** [review]
I generated the freeze files via:

```bash
cabal v2-update
cabal v2-freeze $([[ $(ghc --numeric-version) == 8.6.5 ]] && echo --constraint=any.hsc2hs==0.68.6)
sed -i \
  -e '/abcBridge/d' \
  -e '/cryptol-saw-core/d' \
  -e '/regex-posix/d' \
  -e '/saw-remote-api/d' \
  -e '/saw-script/d' \
  -e 's/ +unix//' \
  -e '/any\.unix/d' \
  -e '/unix-compat/d' \
  cabal.project.freeze
mv -v cabal.project.freeze cabal.GHC-$(ghc --numeric-version).config
```

@atomb The hsc2hs version override is in there because I noticed that you [manually locked](https://github.com/GaloisInc/saw-script/pull/1025/commits/028f34b347ee54d5bbb7678204ae15c681df5965#diff-5c0d57a88cbf6648061b7c0538ae8ec930843e3edf0fa735c29aa60be51e10acL164-R164) it to that version in #1025

Fixes [nightly](https://github.com/GaloisInc/saw-script/actions/runs/577569314) and the [release (build.yml)](https://github.com/GaloisInc/saw-script/actions/runs/577532732) workflows

**Comment by LisannaAtGalois** [comment]
Added an input to the nightly workflow to allow publishing the docker images when triggered manually. Triggered https://github.com/GaloisInc/saw-script/actions/runs/579001044 which will publish them when it completes.

**Comment by pnwamk** [comment]
> Added an input to the nightly workflow to allow publishing the docker images when triggered manually. Triggered https://github.com/GaloisInc/saw-script/actions/runs/579001044 which will publish them when it completes.

It seems to have [skipped push](https://github.com/GaloisInc/saw-script/runs/1929338388?check_suite_focus=true#step:5:10144) (and I don't see an update on [DockerHub](https://hub.docker.com/r/galoisinc/saw-remote-api/tags?page=1&ordering=last_updated)). Am I looking in the wrong spot? Or did something fail to trigger?

**Comment by LisannaAtGalois** [comment]
> > Added an input to the nightly workflow to allow publishing the docker images when triggered manually. Triggered https://github.com/GaloisInc/saw-script/actions/runs/579001044 which will publish them when it completes.
> 
> It seems to have [skipped push](https://github.com/GaloisInc/saw-script/runs/1929338388?check_suite_focus=true#step:5:10144) (and I don't see an update on [DockerHub](https://hub.docker.com/r/galoisinc/saw-remote-api/tags?page=1&ordering=last_updated)).
> 
> ```
> ...
> 10143  Successfully tagged galoisinc/saw-remote-api:nightly
> 10144  Skipping push
> ```
> 
> Am I looking in the wrong spot? Or did something fail to trigger?

Whoops, it was missing some logic. https://github.com/GaloisInc/saw-script/runs/1930549643?check_suite_focus=true

**Comment by LisannaAtGalois** [comment]
@pnwamk Verified that the nightly images are now updated on DockerHub from the manually triggered nightly job against this branch, this should be ready to go in barring any concerns about the freeze files

**Review by pnwamk — APPROVED** [review]

**Comment by LisannaAtGalois** [comment]
A few additional things that would be good to follow up on:
- cryptol doesn't have all of its dependencies frozen, like `libBF`, inviting uncontrolled breakages in compilation whenever these unfrozen dependencies update
- standardize and automate more of the freeze file generation and dependency updating
- building cryptol as part of the pre-merge CI to ensure updates to it don't break nightly / release workflows

**Pull request merged**